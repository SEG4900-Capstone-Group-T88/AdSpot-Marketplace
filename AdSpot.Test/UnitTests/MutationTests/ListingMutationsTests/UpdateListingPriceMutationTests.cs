using AdSpot.Models;

namespace AdSpot.Test.UnitTests.ListingMutationsTests;

[Collection("adspot-inmemory-db")]
public class UpdateListingPriceMutationTests
{
    private const string UpdateListingPriceMutation = """
            mutation UpdateListingPrice($input: UpdateListingPriceInput!) {
                updateListingPrice(input: $input) {
                    listing {
                        listingId
                        price
                    }
                    errors {
                        ... on Error {
                            __typename
                            message
                        }
                    }
                }
            }
        """;

    [Fact]
    [Trait("Category", "Unit")]
    public async Task UpdateListingPriceSuccessful()
    {
        var result = await TestServices.ExecuteRequestAsync(
            scope =>
            {
                var context = scope.ServiceProvider.GetRequiredService<AdSpotDbContext>();
                context.Connections.Add(
                    new Connection
                    {
                        UserId = TestDatabase.TestUser.UserId,
                        PlatformId = TestDatabase.Platforms.First().PlatformId,
                        Handle = "TestAccountHandle",
                        Token = "TestApiToken",
                    }
                );
                context.SaveChanges();

                context.Listings.Add(
                    new Listing
                    {
                        ListingId = 1,
                        PlatformId = TestDatabase.Platforms.First().PlatformId,
                        UserId = TestDatabase.TestUser.UserId,
                        ListingTypeId = TestDatabase
                            .ListingTypes.First(listingType =>
                                listingType.PlatformId == TestDatabase.Platforms.First().PlatformId
                            )
                            .PlatformId,
                        Price = 100.00M,
                    }
                );
                context.SaveChanges();
            },
            b =>
                b.SetQuery(UpdateListingPriceMutation)
                    .SetVariableValue(
                        "input",
                        new Dictionary<string, object?>
                        {
                            { "listingId", 1 },
                            { "price", 120.00M },
                            { "userId", TestDatabase.TestUser.UserId },
                        }.AsReadOnly()
                    )
        );

        result.MatchSnapshot();
    }
}
