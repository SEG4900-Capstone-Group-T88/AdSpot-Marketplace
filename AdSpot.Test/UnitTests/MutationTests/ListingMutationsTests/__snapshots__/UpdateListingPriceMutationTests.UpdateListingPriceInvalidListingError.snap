{
  "data": {
    "updateListingPrice": {
      "listing": null,
      "errors": [
        {
          "__typename": "InvalidListingIdError",
          "message": "Listing with id `2` does not exist."
        }
      ]
    }
  }
}